scripts/eslint-plugin/rules/no-invalid-todo.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+const TODO_REGEX = /\*?\s+(TODO|todo)/;
+const VALID_TODO_REGEX = /\*?\s+(TODO|todo) \[((#\d+)|(\S+\/\S+#\d+)|(W-\d+))\]:/;
+
+/**
+ * This eslint rule checks that all the TODO comments contains a reference to a git issue or a GUS
+ * bug.
+ *
+ * Valid comment formats:
+ *   - `// TODO [#1234]: This is a todo`
+ *   - `// TODO [salesforce/observable-membrane#1234]: This is a todo`
```

Comment:
Should we enforce git references to always contain the full link? This is relevant especially when the Todos are related to the work in other repos.

Reply:
The `org/repo#issue` is enough to reference the git issue. Adding the entire URL will make the comment really long an unpleasant to read.

Reply:
that is fine. What i'm asking is to disallow [#1234] format and to instead always ask for the org/repo#issue

Reply:
I see what you mean, I would like to keep the `org` and `repo` implicit if not necessary. 99% of the `TODO`s are referencing the `salesforce/lwc` repo.

Reply:
Sounds good.